### DISCUSSION TITLE
possible to install sqlpage server on Raspberry pi ?

### DISCUSSION THREAD

**Discussion by schanthery** [sqlpage/SQLPage#283] [Q&A] [answered]
Hello
Is it possible to install and execute sqlpage server on Raspberry pi?
Seb

**Comment by lovasoa** [comment, marked as answer]
Hello and welcome to SQLPage !

SQLPage offers precompiled Docker images for ARM processors, making it the easiest way to use SQLPage with a Raspberry Pi. To set up Docker on an RPi, follow these steps:

1. Install Docker on your Raspberry Pi. You can do this by running the following command:
```bash
curl -sSL https://get.docker.com | sh
```
2. After Docker is installed, you'll need to add your user to the Docker group to run Docker commands without sudo:
```bash
sudo usermod -aG docker your_username
```
Replace `your_username` with your actual username.

3. Log out and log back in for the changes to take effect.

4. Once the image is pulled, you can run SQLPage as a Docker container using:
```bash
docker run -it --name sqlpage -p 8080:8080 --volume "$(pwd):/var/www" --rm lovasoa/sqlpage
```
This command will start SQLPage in a Docker container, mapping port 80 of the container to port 80 of the Raspberry Pi.

6. Access SQLPage by navigating to `http://your_raspberry_pi_ip:8080` in a web browser.

#### references
 - https://github.com/lovasoa/SQLpage?tab=readme-ov-file#with-docker
 - https://hub.docker.com/r/lovasoa/sqlpage/tags

**Reply by schanthery** [reply]
Génial ! Cela fonctionne parfaitement  (le site s'affiche) à un détail important: 
J'ai établi lancer le container sqlpage avec un mappage pour le site avec :

```
docker run -it \
        --name sqlpage \
        -p 8080:8080 \
        --mount type=bind,source=/var/www/sqlpage/www/,target=/var/www/ \
        --mount type=bind,source=/var/www/sqlpage/config/,target=/etc/sqlpage/ \
        --rm lovasoa/sqlpage
```

Mais j'ai un problème de droit d'écriture : sqlpage dans le container docker n'arrive pas à modifier ma base données car il n'a pas les droits dessus. J'ai tenté une modification des droits  mais rien n'y fait mes connaissances sur docker sont trop confuses
Merci pour votre aide
seb

**Reply by lovasoa** [reply]
Si vous avez un fichier de base de données que vous aviez déjà créé avec votre utilisateur local sur le serveur, vous pouvez demander à docker de lancer SQLPage avec le même id d'utilisateur pour y avoir accès, avec l'option `--user`

https://docs.docker.com/engine/reference/run/#user

https://stackoverflow.com/questions/71501506/userid-and-group-id-in-docker-run-command
